python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -183,6 +193,13 @@ def write(
             self.arrow_parquet_args_fn, **self.arrow_parquet_args
         )
         user_schema = write_kwargs.pop("schema", None)
+        # Extract partitioning_flavor before the closure to preserve it across retries
```

Comment:
I felt confused by this comment. What does extracting `partition_flavor` have to do with retries?